CMakeLists.txt

```diff
@@ -35,14 +35,32 @@ endif()
 
 option(RUY_PROFILER "Enable ruy's built-in profiler (harms performance)" OFF)
 
+option(RUY_ENABLE_INSTALL "Enable install rule" ${RUY_IS_TOPLEVEL})
+
 include(cmake/ruy_add_all_subdirs.cmake)
 include(cmake/ruy_cc_library.cmake)
 include(cmake/ruy_cc_binary.cmake)
 include(cmake/ruy_cc_test.cmake)
 
+include(CMakeDependentOption)
+
+option(RUY_USE_EXTERNAL_CPUINFO "Use external cpuinfo" OFF)
+
+cmake_dependent_option(RUY_FIND_CPUINFO "Use find_package to find cpuinfo" ON "RUY_USE_EXTERNAL_CPUINFO" OFF)
+
 # Skip cpuinfo if it was already generated, which can happen when ruy is
 # a subdirectory in a wider project that already uses cpuinfo.
-if (NOT TARGET cpuinfo)
+if (RUY_USE_EXTERNAL_CPUINFO)
```

Comment:
The existing comment, left unchanged in this diff, referred specifically to the existing `if (NOT TARGET cpuinfo)` and is now out of place with the new `if (RUY_USE_EXTERNAL_CPUINFO)`.

Question: if `TARGET cpuinfo` is already `TRUE`, do you still want to go into this `if (RUY_USE_EXTERNAL_CPUINFO)`? If not, maybe you could reinstate the `if (NOT TARGET cpuinfo)`  and move the `if (RUY_USE_EXTERNAL_CPUINFO)` inside it? That would make it easier to review this code to ensure that it is a no-op for existing users.

Reply:
I aimed to provide a more explicit control over the behavior but you're correct that this changes the behavior for existing users which may be undesirable. I have reinstated (and updated) the original condition.

Reply:
Thanks, I can more easily make sense for it now. And I've tried it out, it works for me.